ui/apps/platform/src/Containers/Compliance/Dashboard/useComplianceRunStatuses.ts

```diff
@@ -0,0 +1,108 @@
+import { useState } from 'react';
+import { gql, useApolloClient, useQuery } from '@apollo/client';
+
+import { resourceTypes } from 'constants/entityTypes';
+import {
+    AGGREGATED_RESULTS_ACROSS_ENTITY,
+    AGGREGATED_RESULTS_STANDARDS_BY_ENTITY,
+} from 'queries/controls';
+import {
+    CLUSTERS_COUNT,
+    DEPLOYMENTS_COUNT,
+    NAMESPACES_COUNT,
+    NODES_COUNT,
+} from './ComplianceDashboardTile';
+
+export type ComplianceRunStatusResponse = {
+    complianceRunStatuses: {
+        runs: { state: string }[];
+    };
+};
+
+export function isCurrentScanIncomplete(
+    runs: ComplianceRunStatusResponse['complianceRunStatuses']['runs']
+) {
+    return runs.some((run) => run.state !== 'FINISHED');
+}
```

Comment:
What is the advantage of exporting this checker function, and making calling clients do their own checks, vs. having the response objects include the result of this check "pre-calculated" for the calling client?

Reply:
Good point, I don't think there really is an advantage, and it would be cleaner to return the calculated value as part of the hook signature.